### PR TITLE
FIX: specifically check for vlen strings for string decoding

### PR DESCRIPTION
**Pull request by kmuehlbauer**
See #84 for in-depth analysis.

### CODE REVIEW

**Comment by kmuehlbauer** [comment]
@shoyer So the base problem is that netcdf4 does not decode the `bytes_var` in the linked example notebook even though it is a string (but a fixed length string). The new behaviour especially tests for VLEN strings and only decodes those and let the fixed length strings as is. 

I've also found small differences in some of the special nectdf items (dimnsions etc.) and tried to move the current implementation to be as similar as possible. The only thing left is the NULLTERM vs. NULLPAD issue which I thin has no real (easy) solution.

**Comment by kmuehlbauer** [comment]
Seems, that I'm pretty much lost in the `encoding` rabbit hole.  Decoding all strings might still be correct, though. I'm not really sure, where I'm thinking wrong, but that's obviously the case.... :disappointed: 

If there is a problem with 0.9.0 we might find out soon enough....

**Comment by kmuehlbauer** [comment]
I'm closing this for now, since there is a solution ahead downstream.

**Comment by kmuehlbauer** [comment]
@shoyer So, if we want to fix this issue here, we would have to apply this fix.

Unfortunately then `decode_strings` would need third meaning, "decode all but vlen char" or something like this.

File: h5netcdf/core.py [hunk 1]
```diff
@@ -557,11 +558,11 @@ def _create_dim_scales(self):
                 if self._dim_sizes[dim] is None:
                     kwargs["maxshape"] = (None,)
                 self._h5group.create_dataset(
-                    name=dim, shape=(size,), dtype="S1", **kwargs
+                    name=dim, shape=(size,), dtype=">f4", **kwargs
```
Comment (shoyer) [inline h5netcdf/core.py:561]: To clarify: these other dtype changes are unrelated to the vlen string stuff, right? I assume these are for consistency with netCDF?

**Comment by shoyer** [comment]
> Unfortunately then `decode_strings` would need third meaning, "decode all but vlen char" or something like this.

Maybe we should just rename the option to `decode_vlen_strings`? That's all we need for compatibility with netCDF4.

Then I can issue another release. `decode_strings` hasn't been out in the world for long, so I think it's OK to remove it.

File: h5netcdf/core.py (hunk 1 above)
Comment (kmuehlbauer) [reply h5netcdf/core.py:561]: Yes, I tried to be as consistent with netcdf4 created files.

**Comment by kmuehlbauer** [comment]
> Maybe we should just rename the option to `decode_vlen_strings`? That's all we need for compatibility with netCDF4.
> 
> Then I can issue another release. `decode_strings` hasn't been out in the world for long, so I think it's OK to remove it.

Would you consider having `decode_strings='vlen'` in addition to `decode_strings=True` and `decode_strings=False`? I'm just asking, because that way we would keep the possibility to decode all strings for other `h5netcdf` purposes.

**Comment by shoyer** [comment]
> Would you consider having `decode_strings='vlen'` in addition to `decode_strings=True` and `decode_strings=False`? I'm just asking, because that way we would keep the possibility to decode all strings for other `h5netcdf` purposes.

We could do this, but netCDF's character type has a very weird representation in HDF5, restricted to strings with a fixed width of 1 bytes. As you noticed for Xarray, h5py's string decoding doesn't decode them properly.

We could potentially include a fixed version of this in h5netcdf (e.g., like netCDF4-python's auto-decoding of strings), but I don't think we'll ever want to apply h5py's string decoding directly in this case.

**Comment by kmuehlbauer** [comment]
OK, then I'll rename `decode_strings` -> `decode_vlen_strings`. I'll push updates soon.

**Comment by kmuehlbauer** [comment]
@shoyer I think this is good to go from my end.
